### PR TITLE
implement open graph tags

### PR DESCRIPTION
**Pull request by thanhduc-epi**

### CODE REVIEW

**Comment by lunchin** [comment]
#180

File: src/Foundation/Helpers/HtmlHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                    {
+                        openGraphPage.ExpirationTime = foundationPageData.StopPublish.Value;
+                    }
+
+                    openGraphOutput = helper.OpenGraph(openGraphPage).ToString();
+                    break;
+
+                case EntryContentBase entryContentBase:
+                    var openGraphEntry = new OpenGraphProduct(entryContentBase.DisplayName, new OpenGraphImage(_assetUrlResolver.Value.GetAssetUrl(entryContentBase)), GetUrl(entryContentBase.ContentLink));
+
```
[the remark this answers is not in the document]
Comment (thanhduc-epi) [reply src/Foundation/Helpers/HtmlHelpers.cs]: For "availability" and "condition",  I did not see any existing properties of the product for them, not sure what you really mean here. Should I think about a scenario to create the business logic here?

File: src/Foundation/Infrastructure/OpenGraph/Idio/OpenGraphFoundationPageData.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+﻿using Boilerplate.Web.Mvc.OpenGraph;
+using Foundation.Infrastructure.OpenGraph.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Foundation.Infrastructure.OpenGraph.Idio
```
Comment (lunchin) [inline src/Foundation/Infrastructure/OpenGraph/Idio/OpenGraphFoundationPageData.cs]: Can we remove the idio namespace

File: src/Foundation/Infrastructure/OpenGraph/Idio/OpenGraphGenericNode.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
+﻿using Boilerplate.Web.Mvc.OpenGraph;
+using Foundation.Infrastructure.OpenGraph.Extensions;
+using System;
+using System.Text;
+
+namespace Foundation.Infrastructure.OpenGraph.Idio
```
Comment (lunchin) [inline src/Foundation/Infrastructure/OpenGraph/Idio/OpenGraphGenericNode.cs]: Can we remove the idio namespace

File: src/Foundation/Infrastructure/OpenGraph/Idio/OpenGraphGenericProduct.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Boilerplate.Web.Mvc.OpenGraph;
+using Foundation.Infrastructure.OpenGraph.Extensions;
+
+namespace Foundation.Infrastructure.OpenGraph.Idio
```
Comment (lunchin) [inline src/Foundation/Infrastructure/OpenGraph/Idio/OpenGraphGenericProduct.cs]: Can we remove the idio namespace

File: src/Foundation/Infrastructure/OpenGraph/Idio/OpenGraphHomePage.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using Boilerplate.Web.Mvc.OpenGraph;
+
+namespace Foundation.Infrastructure.OpenGraph.Idio
```
Comment (lunchin) [inline src/Foundation/Infrastructure/OpenGraph/Idio/OpenGraphHomePage.cs]: Can we remove the idio namespace

File: src/Foundation/Infrastructure/OpenGraph/Idio/OpenGraphLocationItemPage.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
+﻿using Boilerplate.Web.Mvc.OpenGraph;
+using Foundation.Infrastructure.OpenGraph.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Foundation.Infrastructure.OpenGraph.Idio
```
Comment (lunchin) [inline src/Foundation/Infrastructure/OpenGraph/Idio/OpenGraphLocationItemPage.cs]: Can we remove the idio namespace

**Pull request merged**